### PR TITLE
use Codespaces secrets for certs

### LINKED ISSUE #211: Key vault and certs setup when using AIRS subscription
When folks are deploying in their own subscriptions, what is a good way to give them access to the key vault that has the certificates?

### PR DESCRIPTION
**Pull request by bartr**
# Type of PR

- [x] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

Description
- add docs and scripts to setup the baseline in airs subscriptions
- update setup.sh script to work with the new template changes

## Issues Closed or Referenced

- Closes #211 
- Closes https://github.com/retaildevcrews/asb-hack/issues/171

### CODE REVIEW

**Review by AAkindele — COMMENTED** [review]
I was able to get airs-setup.sh working after making some manual changes locally to environment variables for certs and traefik template. I wanted to verify some assumptions before making changes to this branch directly. Here is a diff of the changes I made with notes. https://github.com/retaildevcrews/asb-hack/compare/airs...aakindele/airs

**Comment by bartr** [comment]
I may have set the secrets incorrectly when I recreated them. I'll double check. 

I would manually set the env vars and get it working and then validate against the env vars once updated.

**Comment by AAkindele** [comment]
I tested the latest changes to setup and cleanup scripts and they work! Changes have been made to cluster-stamp.json in the main branch. So next steps is to get those changes into this branch and test again.

File: airs-setup.sh [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+# make sure the locations are different
+if [ "$ASB_LOCATION" == "$ASB_GEO_LOCATION" ]
+then
+  echo ASB_LOCATION and ASB_GEO_LOCATION must be different regions
+  echo Using paired regions is recommended
+  exit 1
+fi
+
+# AAD admin group name
```
Comment (bartr) [inline airs-setup.sh:65]: we should remove this default and make it a required param

File: setup.sh [hunk 2]
```diff
@@ -202,6 +201,9 @@ az keyvault set-policy --certificate-permissions get --object-id $ASB_POD_MI_ID
 az keyvault set-policy --secret-permissions get --object-id $ASB_POD_MI_ID -n $ASB_KV_NAME
 
 # config traefik
```
Comment (bartr) [inline setup.sh:203]: can we change the name from "manual-test"?

File: templates/cluster-stamp.json [hunk 3]
```diff
@@ -37,8 +37,16 @@
         },
         "aksIngressControllerCertificate": {
             "type": "string",
```
Comment (bartr) [inline templates/cluster-stamp.json:39]: we should be able to delete defaultValue line

File: templates/cluster-stamp.json [hunk 4]
```diff
@@ -37,8 +37,16 @@
...
+            "defaultValue": "",
             "metadata": {
-                "description": "The base 64 encoded AKS Ingress Controller public certificate (as .crt or .cer) to be stored in Azure Key Vault as secret and referenced by Azure Application Gateway as a trusted root certificate."
+                "description": "The AKS Ingress Controller public certificate (as .crt or .cer) to be stored in Azure Key Vault as secret and referenced by the AKS ingress controller."
+            }
+        },
+        "aksIngressControllerKey": {
+            "type": "string",
```
Comment (bartr) [inline templates/cluster-stamp.json:46]: should be able to delete defaultValue

File: setup.sh (hunk 2 above)
Comment (AAkindele) [reply setup.sh:200]: 👍 i copied the value to a new secret called 'aks-sb-crt'. testing it in my latest deployment.

**Comment by jofultz** [comment]
I'm reviewing from my phone, but realize that I have edits. I'll make edits and comments in the morning.

File: README-AIRS.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
+# AKS Secure Baseline Hack (internal)
+
+## Welcome to the Patterns and Practices (PnP) AKS Secure Baseline (ASB) hack!
+
+- This repo contains modifications and additions to tailor it specifically for hack events and **should not be used for production deployments**
+  - Original PnP AKS Secure Baseline repo is located [here](https://github.com/mspnp/aks-secure-baseline)
```
Comment (jofultz) [inline README-AIRS.md:6]: I modified order, format, and slight word changes for the first bullet and subbullet. Please check.

File: README-AIRS.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  az group list -o table | grep $ASB_TEAM_NAME
+
+  # make sure the branch doesn't exist
+  git branch -a | grep $ASB_TEAM_NAME
+
+  # if either exists, choose a different team name and try again
+else 
+    echo "Team name doesn't match required format: [starts with a-z, [a-z,0-9], max length 8]" 
+fi
+
```
Comment (jofultz) [inline README-AIRS.md:75]: I added this early validation check for the team name.  Please check the logic.

**Review by jofultz — APPROVED** [review]
@bartr & @AAkindele, I made a couple of updates and put comments on them.  Please check my changes.  I'm approving as it is now, but I'm okay if you revert to the original.

**Pull request merged**